addons/portal/tests/test_portal_wizard.py

```diff
@@ -145,16 +145,15 @@ def test_portal_wizard_error(self):
         portal_user = portal_wizard.user_ids
 
         self.internal_user.login = 'test_error@example.com'
-        portal_user.email = 'test_error@example.com'
 
-        with self.assertRaises(UserError, msg='Must detect the already used email.'):
-            portal_user.action_revoke_access()
+        portal_user.email = 'test_error@example.com'
+        self.assertEqual(portal_user.email_validity_state, 'already_registered', msg='Must detect the already used email.')
 
         portal_user.email = 'wrong email format'
-        with self.assertRaises(UserError, msg='Must detect wrong email format.'):
-            portal_user.action_revoke_access()
+        self.assertEqual(portal_user.email_validity_state, 'invalid', msg='Must detect wrong email format.')
```

Comment:
If we keep the raise, maybe we could keep this code?
We could check both actually, the value of "email_validity_state" AND the fact that it raises when we try to call the action.